### DISCUSSION TITLE
Schema Validation Throw An Error.

### DISCUSSION THREAD

**Discussion by Vishwas-verma** [adonisjs/core#2666] [General]
**Here is my code in controller-**

```
public async signup({request, response}: HttpContextContract) {
    try {
       const inputData = await request.validate({
         schema: signupSchema
       })
      const user = await User.create(inputData)
      return response.json({
        user : await (new UserTransformer()).transform(user),
      })
    } catch (e) {
      response.badRequest(e.message)
    }
  }
```

  **Here is my signup schema** 

```
import {GenderEnum} from "App/Enums/gender.enum";
import {rules, schema} from "@ioc:Adonis/Core/Validator";

export const signupSchema = schema.create({
  first_name: schema.string(),
  last_name: schema.string.optional(),
  gender: schema.enum([GenderEnum.MALE, GenderEnum.FEMALE, GenderEnum.TRANSGENDER]),
  phone_number: schema.number.optional(),
  email: schema.string({}, [
    rules.email(),
    rules.unique({table: 'users', column: 'email'})
  ]),
  password: schema.string({}, [
    rules.minLength(8),
  ])
})
```
 **Input Request**
```
{
    "first_name":"Vishwas",
    "last_name":"verma",
    "gender":"male",
    "email":"random@gmail.com",
    "password":"loremipsum"
}
```
  **I am getting validation issue-**

E_VALIDATION_FAILURE: Validation Exception

Below is the snippet request-response image-
![Screenshot from 2021-05-24 16-49-49](https://user-images.githubusercontent.com/41827033/119340439-240c8b80-bcb0-11eb-962e-2a099c64b9e0.png)

This issue basically arises after installation  of auth .
Can anyone help me with it.

Thanks in advance.

**Comment by mastermunj** [comment]
Couple of inputs to make your problem statement easy to understand:
* Please format the code blocks properly.
* Please provide the input request data for which you are getting the error.
* Please provide the whole error object instead of just the message to understand what type of validation errors are there.

**Reply by Vishwas-verma** [reply]
Hye @mastermunj I have updated the discussion please rectify if any error is there.

**Reply by mastermunj** [reply]
Please console log the error inside try catch block to check what's the validation error returned. You can share it here as well for us to debug further.

Also, you shouldn't simply send the e.message if the validation fails. Instead, send the whole error back so that Adonis can process the error in proper way and send appropriate response.

**Reply by Vishwas-verma** [reply]
Sending whole error object worked instead of sending e.message.

**Reply by Vishwas-verma** [reply]
@mastermunj 
https://github.com/adonisjs/core/discussions/2669 -Please have a look at this discussion
